### DISCUSSION TITLE
Cannot provide passphrase with `age -d`

### DISCUSSION THREAD

**Discussion by gabyx** [FiloSottile/age#572] [UX reports]
For decrypting a passphrase encrypted file with `age`, its not possible at the moment to provide the passphrase through a file or stdin.

```shell

echo "asdf" | age -d secret.age
```

will still prompt on the `/dev/tty` for a passphrase. Is there currently a workaround or otherwide it would be nice to have the 

`--passphrase <file>` or `--passphrase -` (stdin) command arguments available.

**Comment by zhuoqun-chen** [comment]
I need this feature too!!!

**Comment by psanford** [comment]
You can use a small `expect` script if you really need this behavior:
```
 #!/usr/bin/env expect

# Check if the correct number of arguments is provided
if {$argc != 2} {
    send_user "Usage: $argv0 <password> <encrypted_file>\n"
    exit 1
}

# Get the password and file from command-line arguments
set password [lindex $argv 0]
set file [lindex $argv 1]

# Spawn the age command
spawn age -d $file

# Wait for the password prompt
expect "Enter passphrase: "

# Send the password followed by a newline
send "$password\r"

# Wait for the command to finish and capture its output
expect {
    eof {
        # Command completed, display its exit status
        catch wait result
        exit [lindex $result 3]
    }
    timeout {
        send_user "Error: Command timed out\n"
        exit 1
    }
}
```

**Reply by zhuoqun-chen** [reply]
very helpful! thx @psanford !

**Comment by gabyx** [comment]
Actually the way to go by the author (if I am not mistaken) is not use a passphrase at all:

Why because its unsafe, better is to just generate a `age-keygen` pair and use the public key to encrypt the REAL private keyfile, the password to decrypt your private key is then just `echo "your private key..." | age -d -i - my-real-private-key.age`.

@FiloSottile : Is that correct?

**Comment by haf** [comment]
So instead of using encryption because of this missing feature that has been three years in the making, I'm now saving the files in plaintext instead! Good usability!

**Comment by clach04** [comment]
Cross-reference to https://github.com/FiloSottile/age/discussions/256

@gabyx  I recommend closing this thread as it duplicates other threads.

https://github.com/FiloSottile/age/pull/520 works for me

**Comment by mirql** [comment]
here example with "expect", without compilation or other pains

https://github.com/FiloSottile/age/pull/520#issuecomment-2768572661

**Comment by AlexanderYastrebov** [comment]
Related #603 and #641

**Comment by FiloSottile** [comment]
The new batchpass plugin handles non-interactive passphrase encryption. See https://github.com/FiloSottile/age/discussions/256#discussioncomment-15334934 for usage and a warning.
